File: src/ui/app/root/application/models/HumanResource/Staff.ts
export interface IHumanResourceStaff {
	staffId : string;
	departmentId : string;
	designationId : string;
	countryId : string;
	bloodGroupId : string;
	religionId : string;
	genderId : string;
	fullName : string;
	fatherName : string;
	cnic : string;
	dateOfBirth : Date;
	email : string;
	maritalStatus : boolean;
	contactNo : string;
	address : string;
	picture : string;
	statusId : number;
	loggerId : string;
	empolyeeCode : string;
	cityId : string;
	subCityId:string;

}

export interface IProfileStaff {
    staffId: string;
    departmentId: string;
    designationId: string;
    genderId: string;
    fullName: string;
    fatherName: string;
    dateOfBirth: Date;
    email: string;
    maritalStatus: boolean;
    contactNo: string;
    address: string;
    picture: string;
    statusId: number;
    loggerId: string;
	userId: number;
	countryId : string;
	bloodGroupId : string;
	religionId : string;
	cNIC:string;
}

export interface StaffHODData {
    ide: string;
	fullName: string;
	email:string;
	isChecked:boolean;
}
export interface teachercheckmodel {
   
	fullName: string;
	email:string;
	
}
export interface IHumanResourceEvaluationMaster {

    evaluationId: string;
    fullName: string;
	description: string;
	popupDescription: string;
	statusId:number
}



export interface IHumanResourceEvaluationDetail {

    evaluationDetailId: string;
    evaluationId: string;
	options: Array<optionsls>;
	statusId:number;
	questions:string;
}



export interface optionsls {

    order: number;
    option: string;

}

export interface IEvaluationMaster {
	evaluationId: string;
	fullName: string;
	description: string;
	statusId: number;


}

export interface EvaluationDetail {
	evaluationDetailId: string;
	fullName: string;
	description: string;
	statusId: number;
	options: Array<optionsls>;

	questions: string;


}
export interface ITeacherProfileList {
    newID: string;
    teacherId: string;
    teacherName: string;
    email: string;
    cNIC: string;
    contactNo: string;
    department: string;
    designation: string;
    cityName: string;
    subCity: string;
}
export interface IStaffByCampus {
	staffId : string;
	fullName : string;
	fatherName : string;
	email : string;
	}

File: src/data/Model/Cms360/HumanResource/Staff.cs
/*
*   Auther: H.Muhammad Kamran
*   email: hmuhdkamran@gmail.com
*   contact: +92 (313 / 333) 9112 845
*/

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cms360.Data.Model
{
    [Table("Staff", Schema = "HumanResource")]
    public partial class HumanResourceStaff
    {
        [Key]
        [Required]
        public Guid StaffId { get; set; }

        public Guid DepartmentId { get; set; }

        public Guid DesignationId { get; set; }
        public Guid CountryId { get; set; }
        public Guid BloodGroupId { get; set; }
        public Guid ReligionId { get; set; }

        public Guid GenderId { get; set; }

        public string FullName { get; set; }

        public string FatherName { get; set; }

        public string CNIC { get; set; }

        public DateTime DateOfBirth { get; set; }
        public string Email { get; set; }

        public bool MaritalStatus { get; set; }
        [Column(TypeName = "jsonb")]
        public string ContactNo { get; set; }
        [Column(TypeName = "jsonb")]
        public string Address { get; set; }
        public string Picture { get; set; }

        public int StatusId { get; set; }

        public Guid LoggerId { get; set; }

        public string EmpolyeeCode { get; set; }
        public Nullable<Guid> CityId { get; set; }
        public Nullable<Guid> SubCityId { get; set; }


    }

    public class StaffHODData
    {
        
        [Key]
         public Guid Ide { get; set; }

         public string FullName {get;set;}

         public string Email { get; set; }

         public bool IsChecked { get; set; }

    }
    public  class EvaluationMaster
    {
        [Key]
         public Guid EvaluationId { get; set; }

         public string FullName {get;set;}

         public string Description { get; set; }

         public int StatusId { get; set; }


    }
    public  class EvaluationDetail
    {
        [Key]
         public Guid EvaluationDetailId { get; set; }

         public string FullName {get;set;}

         public string Description { get; set; }

         public int StatusId { get; set; }
         [Column(TypeName = "jsonb")]

         public string Options  {get;set;}

         public string Questions {get;set;}


    }

    public  class TeacherProfileList
    {
        [Key]
         public Guid NewID { get; set; }

         public Guid TeacherId {get;set;}

         public string TeacherName { get; set; }

         public string Email { get; set; }

         public string CNIC { get; set; }

        [Column(TypeName = "jsonb")]
         public string ContactNo { get; set; }

         public string Department { get; set; }

         public string Designation { get; set; }

         public string CityName { get; set; }

         public string SubCity { get; set; }
    }


        public  class StaffByCampus
    {
        [Key]

         public Guid StaffId {get;set;}
         public string FullName { get; set; }

         public string Email { get; set; }
         public string FatherName { get; set; }

      
    }

}